Core/src/org/sleuthkit/autopsy/casemodule/services/NewUserTagNameDialog.java

```diff
@@ -0,0 +1,241 @@
+/*
+* Autopsy Forensic Browser
+*
+* Copyright 2011-2016 Basis Technology Corp.
+* Contact: carrier <at> sleuthkit <dot> org
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package org.sleuthkit.autopsy.casemodule.services;
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.Toolkit;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import org.openide.util.NbBundle;
+
+public class NewUserTagNameDialog extends javax.swing.JDialog {
+
+    private String userTagDisplayName;
+    private BUTTON_PRESSED result;
+
+    enum BUTTON_PRESSED {
+        OK, CANCEL;
+    }
+
+    /**
+     * Creates a new NewUserTagNameDialog dialog.
+     */
+    NewUserTagNameDialog() {
+        super(new JFrame(NbBundle.getMessage(NewUserTagNameDialog.class, "NewUserTagNameDialog.title.text")),
+                NbBundle.getMessage(NewUserTagNameDialog.class, "NewUserTagNameDialog.title.text"), true);
+        initComponents();
+        String test = "";
+        this.display();
+    }
+
+    /**
+     * Sets display settings for the dialog and adds appropriate listeners.
+     */
+    private void display() {
+        setLayout(new BorderLayout());
+
+        /*
+         * Center the dialog 
+         */
+        Dimension screenDimension = Toolkit.getDefaultToolkit().getScreenSize();
+        int width = this.getSize().width;
+        int height = this.getSize().height;
+        setLocation((screenDimension.width - width) / 2, (screenDimension.height - height) / 2);
+
+        /*
+         * Add a handler for when the dialog window is closed directly.
+         */
+        this.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e) {
+                doButtonAction(false);
+            }
+        });
+
+        /*
+         * Add a listener to enable the save button when the text field changes.
+         */
```

Comment:
One other thing, I think it would be better if the menu presentatino code decide what tag names to use to populate the menu rather than the TagsManager, i.e., I think we should replace TagManager.getAllTagNamesForDisplay with a getUserTagNames metyhod (and other methods, if necessary) and make the menu presenter decide how to merge the various sets of tags names for the menu. In other words, we should have a clear separation of concerns between the service maintaining the tags, the "business layer" of the application, and the "presentation layer" of the application.